triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Converter.scala

```diff
@@ -34,205 +36,133 @@ import com.daml.platform.participant.util.LfEngineToApi.{
 import scala.concurrent.duration.{FiniteDuration, MICROSECONDS}
 
 // Convert from a Ledger API transaction to an SValue corresponding to a Message from the Daml.Trigger module
-case class Converter(
-    fromTransaction: Transaction => Either[String, SValue],
-    fromCompletion: Completion => Either[String, SValue],
-    fromHeartbeat: SValue,
-    fromACS: Seq[CreatedEvent] => Either[String, SValue],
-    toFiniteDuration: SValue => Either[String, FiniteDuration],
-    toCommands: SValue => Either[String, Seq[Command]],
-    toRegisteredTemplates: SValue => Either[String, Seq[Identifier]],
-)
+final class Converter(compiledPackages: CompiledPackages, triggerIds: TriggerIds) {
 
-// Helper to create identifiers pointing to the Daml.Trigger module
-case class TriggerIds(val triggerPackageId: PackageId) {
-  def damlTrigger(s: String) =
-    Identifier(
-      triggerPackageId,
-      QualifiedName(ModuleName.assertFromString("Daml.Trigger"), DottedName.assertFromString(s)),
-    )
-  def damlTriggerLowLevel(s: String) =
-    Identifier(
-      triggerPackageId,
-      QualifiedName(
-        ModuleName.assertFromString("Daml.Trigger.LowLevel"),
-        DottedName.assertFromString(s),
-      ),
-    )
-  def damlTriggerInternal(s: String) =
-    Identifier(
-      triggerPackageId,
-      QualifiedName(
-        ModuleName.assertFromString("Daml.Trigger.Internal"),
-        DottedName.assertFromString(s),
-      ),
-    )
-}
-
-object Converter {
+  import Converter._
   import com.daml.script.converter.Converter._, Implicits._
 
-  private case class AnyContractId(templateId: Identifier, contractId: ContractId)
-  private case class AnyTemplate(ty: Identifier, arg: SValue)
-  sealed abstract class AnyChoice extends Product with Serializable {
-    def name: ChoiceName
-    def arg: SValue
-  }
-  object AnyChoice {
-    final case class Template(name: ChoiceName, arg: SValue) extends AnyChoice
-    final case class Interface(ifaceId: Identifier, name: ChoiceName, arg: SValue) extends AnyChoice
-  }
-  private case class AnyContractKey(key: SValue)
+  private[this] val valueTranslator = new preprocessing.ValueTranslator(
+    compiledPackages.pkgInterface,
+    requireV1ContractIdSuffix = false,
+  )
+
+  private[this] def validateRecord(r: value.Record): Either[String, Value.ValueRecord] =
```

Comment:
Given the method (here and elsewhere) has no state, surely its safe to share this method (i.e. not use the `this` restriction)? Simply thinking its better to share the method code per class than per object.

Reply:
I prefer to use the same annotation consistently, it becomes very quickly tricky to reason about which method has state or not. 
I favored `private[this]` over `private` are it is more restrictive. I would probably be fine to change my default style  and use `private` everywhere, but let's discuss that another time.